react/QuantitySelector.js

```diff
@@ -15,7 +15,7 @@ class QuantitySelector extends Component {
     } else if (quantity < 0) {
       quantity = 0
     }
-    event.target.value = quantity
+    this.setState({ currentQuantity: quantity })
```

Comment:
You also need to use the value on `render`. Set the initial state with `getDerivedStateFromProps`, [check it's documentation](https://reactjs.org/docs/react-component.html#static-getderivedstatefromprops).